qiskit/compiler/assemble.py

```diff
@@ -152,7 +152,7 @@ def assemble(experiments: Union[QuantumCircuit, List[QuantumCircuit], Schedule,
         return assemble_circuits(circuits=bound_experiments, qobj_id=qobj_id,
                                  qobj_header=qobj_header, run_config=run_config)
 
-    elif all(isinstance(exp, ScheduleComponent) for exp in experiments):
+    elif all(isinstance(exp, NamedValue) for exp in experiments):
```

Comment:
Instead, check for 
```suggestion
    elif all(isinstance(exp, [Instruction, Schedule]) for exp in experiments):
```
This will be more maintainable going forward, because the reader(developer) knows that the experiment must be an Instruction or Schedule for this to be true. (If other components in pulse, such as waveform, subclassed from NamedValue, this would fail)

Reply:
Noted, please have a look at #5245 . Moved it to a new PR

Reply:
Set a tuple as a Union isn't acceptable as type check argument